### PR TITLE
Log Warning when secret is detected in DSN

### PR DESCRIPTION
**Pull request by SimonCropp**
quoting @bruno-garcia 

> Secret has been deprecated so many years ago that we can just delete it at this point. The SDK requires the /envelope endpoint so doesn't support old Sentry's that would require a secret in the DSN

https://github.com/getsentry/sentry-dotnet/pull/1747#discussion_r906612700

### CODE REVIEW

File: test/Sentry.Tests/Protocol/DsnTests.cs [hunk 1]
```diff
@@ -133,21 +126,15 @@ public void Ctor_NullDsn_ThrowsArgumentNull()
     }
 
     [Fact]
-    public void TryParse_SampleValidDsnWithoutSecret_Succeeds()
-    {
-        Assert.NotNull(Dsn.TryParse(DsnSamples.ValidDsnWithoutSecret));
-    }
-
-    [Fact]
-    public void TryParse_SampleValidDsnWithSecret_Succeeds()
```
Comment (mattjohnsonpint) [inline test/Sentry.Tests/Protocol/DsnTests.cs:142]: Shouldn't we still keep the Parse and TryParse tests? Though we don't use the secret ourself, it's still valid input from the user.

**Comment by mattjohnsonpint** [comment]
Perhaps we should add some logic that logs a warning message if they pass a DSN that includes a secret?

**Review by bruno-garcia — APPROVED** [review]
Much simpler. Thanks!

File: test/Sentry.Tests/Protocol/DsnTests.cs (hunk 1 above)
Comment (SimonCropp) [reply test/Sentry.Tests/Protocol/DsnTests.cs:142]: @bruno-garcia is this correct. from your statement that spured this, i cant tell if secret is still valid?

Comment (bruno-garcia) [reply test/Sentry.Tests/Protocol/DsnTests.cs:142]: we don't need to validate secret anymore. Test can be deleted.

**Comment by bruno-garcia** [comment]
> Perhaps we should add some logic that logs a warning message if they pass a DSN that includes a secret?

There's no problem with having the secret, it's just not going to be used for anything on the server.

File: test/Sentry.Tests/Protocol/DsnTests.cs (hunk 1 above)
Comment (SimonCropp) [reply test/Sentry.Tests/Protocol/DsnTests.cs:142]: re added the tests for with secret

**Comment by SimonCropp** [comment]
i added a warning when secret is detected

**Comment by mattjohnsonpint** [comment]
Fixed merge with #1758

**Pull request merged**